platform/pulpcore/app/serializers/base.py

```diff
@@ -144,6 +143,39 @@ def _populate_generic_fields(self, instance, field_mappings):
             field.mapping.replace(mapping)
 
 
+class NestedModelSerializer(ModelSerializer):
+    """
+    This class allows the serializer to have WritableNestedUrlRelated fields, which represent
+    nested parent objects that are implied by the url of a request.
+    """
+
+    def to_internal_value(self, *args, **kwargs):
+        """
+        After letting DRF populate and validate all writable fields, we populate and validate
+        `href_writable` fields.
+        """
+        # super().to_internal_value will validate all writable fields and populate serializer.data
+        validated_data = super().to_internal_value(*args, **kwargs)
+        # nested parent fields should be read_only (so they don't accept args that are ignored)
+        # instead, validate the parent based on the parent url and add it to the validated data.
+        validated_data.update(self.writable_nested_parent_url_to_internal())
```

Comment:
Why is `writable_nested_parent_url_to_internal()` a separate method instead of having that code in the body here?
This method does almost nothing else.